runtime/packages/core-modules/assembly/executive.ts

```diff
@@ -47,11 +55,100 @@ export namespace Executive{
     export function finalizeBlock(): Header {
         return System.finalize();
     }
+    /**
+     * creates inherents from internal modules
+     * @param data inherents
+     */
+    export function createExtrinsics(data: InherentData): u8[] {
+        const timestamp: Inherent = Timestamp.createInherent(data);
+        const aura = AuraModule.createInherent(data);
+        if(aura.isSome()){
```

Comment:
I think that it would be better if the `AuraModule.createInherent` function returns empty array. This way you won't need to add this if and the code complexity will be smaller